### PR TITLE
general bug fixes (a collection)

### LINKED ISSUE #159: Fix `like` functionality and document in the user's guide
## 💡 Description

See related bug #134, partly closed

### LINKED ISSUE #162: fields parameter does not return values if more than one value is requested for CSV format
## 🐛 Describe the bug

Request with fields parameter and response format CSV does not return multiple fields.

In addition, in our test example, the lidvid in the response are replicated

## 📜 To Reproduce
1. When I do the request:
```
% curl -G 'https://pds.nasa.gov/api/search/1.0/products' \
    --data-urlencode 'limit=1000' \
    --data-urlencode 'q=pds:Identification_Area.pds:product_class eq "Product_Observational"' \
    --data-urlencode 'fields=pds:Science_Facets.pds:discipline_name' \
    --data-urlencode 'fields=lidvid'  \
    --data-urlencode 'fields=ops:Harvest_Info.ops:node_name' 
    --header 'Accept:text/csv'

```

2. The result only contains the lidvid and empty fields for node_name and discipline:
```
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044883359_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044883559_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044883759_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044883959_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044884159_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044884359_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044884559_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044884759_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044884959_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044885159_csv::1.0",,
"urn:nasa:pds:mess_xrs_derived:data_footprints:xrs_fp_2_044885359_csv::1.0",,
```

3. In addition the result is replicated, only different lidvid should show

## 🕵️ Expected behavior

All the three fields value should be returned.

The lidvid should not be replicated.

## 📚 Version of Software Used

version 1.0.1 in production.

## 🖥 System Info

---

### LINKED ISSUE #167: summary-only does not work as expected
## 🐛 Describe the bug
Related to #152. On the main branch summary-only is signaled via limit=0, and it is being too efficient.

```
$ curl --location --request GET 'http://localhost:8080/uid/urn:nasa:pds:insight_rad:data_derived::7.0/referencing/product?limit=0&fields=lid,lidvid' --header 'Accept: application/kvp+json'
{
  "summary":{"q":"","hits":0,"took":18,"start":0,"limit":0,"sort":[],"properties":[]},
  "data":[
  ]
}
```

Missing hits, and properties. On release 1.0.1 when `Accept: applicaation/json` it also has a null point exception during processing (#152 ) which is a different problem but related.

## 📜 To Reproduce

See above example

## 🕵️ Expected behavior

```
$ curl --location --request GET 'http://localhost:8080/uid/urn:nasa:pds:insight_rad:data_derived::7.0/referencing/product?limit=0&fields=lid,lidvid' --header 'Accept: application/kvp+json'
{
  "summary":{"q":"","hits":7,"took":7,"start":0,"limit":0,"sort":[],"properties":["lid","lidvid"]}
}
```
## 📚 Version of Software Used
main branch and release 1.0.1

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Epic to include several issues

## ⚙️ Test Data and/or Report

fixes NASA-PDS/registry-api#159

api_ver

fixes NASA-PDS/registry-api#162:
```
$ curl --location --request GET 'http://localhost:8080/gid/bundle?fields=lid,lidvid,ops:Data_File_Info.ops:creation_date_time' --header 'Accept: text/csv'
lid,lidvid,ops:Data_File_Info.ops:creation_date_time
"urn:nasa:pds:insight_rad","urn:nasa:pds:insight_rad::2.1","2020-01-03T17:58:09Z"
```

fixes NASA-PDS/registry-api#167
```
$ curl --location --request GET 'http://localhost:8080/uid/urn:nasa:pds:insight_rad:data_derived::7.0/referencing/product?limit=0&fields=lid,lidvid' --header 'Accept: application/json'
{"summary":{"q":"","hits":7,"took":11,"start":0,"limit":0,"sort":[],"properties":[]}}
```

## ♻️ Related Issues
NASA-PDS/registry-api#149
NASA-PDS/registry-api#152 -- fixed with NASA-PDS/registry-api#167 on main branch
NASA-PDS/registry-api#162 
NASA-PDS/registry-api#167

### CODE REVIEW

File: service/src/main/java/gov/nasa/pds/api/registry/model/RefLogicAny.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	private ReferencingLogicTransmuter resolveID (ControlContext context, UserContext input)
+			throws IOException, LidVidNotFoundException, UnknownGroupNameException
+	{
+		return ReferencingLogicTransmuter.getByProductClass(
+				QuickSearch.getValue(context.getConnection(),
+			             LidVidUtils.resolve(input.getIdentifier(), ProductVersionSelector.TYPED, context, RequestBuildContextFactory.empty()),
+			             "product_class"));
+	}
+
+	private RequestAndResponseContext search(ControlContext context, UserContext input, boolean isIdToGroup)
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/model/RefLogicAny.java:37]: @sonatype-lift ignore

File: model/pom.xml [hunk 2]
```diff
@@ -112,27 +112,22 @@
     <!--SpringFox dependencies -->
     <dependency>
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply model/pom.xml:121]: @sonatype-lift ignore

File: service/src/main/java/gov/nasa/pds/api/registry/util/GroupConstraintImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import com.google.errorprone.annotations.Immutable;
+
+import gov.nasa.pds.api.registry.GroupConstraint;
+
+@Immutable
+public class GroupConstraintImpl implements GroupConstraint
+{
+	final private Map<String,List<String>> all;
+	final private Map<String,List<String>> any;
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/util/GroupConstraintImpl.java:15]: @sonatype-lift ignore

File: service/src/main/java/gov/nasa/pds/api/registry/view/PdsProductsTextHtmlSerializer.java [hunk 4]
```diff
@@ -38,7 +38,8 @@ protected void writeInternal(PdsProducts t, HttpOutputMessage outputMessage)
         OutputStream os = outputMessage.getBody();
         OutputStreamWriter wr = new OutputStreamWriter(os);
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/registry/view/PdsProductsTextHtmlSerializer.java]: fixed

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

Looking for a review. Going to wade into harvest now. I think this fixes most of the major bugs and should only be applied to the main branch. I will merge it after review.

**Comment by jordanpadams** [comment]
@al-niessner can we merge this branch with your changes along with the rest of the main branch that got messed up with the force merge?

**Review by jordanpadams — APPROVED** [review]

**Pull request merged**